clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/web/DynamicDataClient.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.clients.simpleclient.web;
+
+import feign.Response;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import uk.gov.gchq.palisade.clients.simpleclient.request.AddSerialiserRequest;
+import uk.gov.gchq.palisade.clients.simpleclient.request.ReadRequest;
+
+/**
+ * The interface Dynamic data client.
+ */
+public interface DynamicDataClient {
```

Comment:
What makes this class 'dynamic'?

Reply:
its dynamic in the sense that the feign builders in urlDataClient or namedDataClient will implement this class and change the content of the feign builder accordingly

Reply:
Could possibly be better named DataClientPrototype or something, but the idea was to differentiate it from the PalisadeClient - this needs to be "built" by specifying the name of a data-service

Reply:
Can this be renamed to DataClientFactory then?